test/functionapp.test.ts

```diff
@@ -0,0 +1,121 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import { WebSiteManagementClient, WebSiteManagementModels } from 'azure-arm-website';
+import { IHookCallbackContext, ISuiteCallbackContext } from 'mocha';
+import * as vscode from 'vscode';
+import { AzureTreeDataProvider, DialogResponses, ext, extensionPrefix, FunctionAppProvider, getRandomHexString, TestAzureAccount, TestUserInput } from '../extension.bundle';
+import { longRunningTestsEnabled } from './global.test';
+
+// tslint:disable-next-line:max-func-body-length
+suite('Function App actions', async function (this: ISuiteCallbackContext): Promise<void> {
+    this.timeout(1200 * 1000);
+    const resourceGroupsToDelete: string[] = [];
+    let oldAdvancedCreationSetting: boolean | undefined;
+    const resourceName: string = getRandomHexString().toLowerCase();
+    const testAccount: TestAzureAccount = new TestAzureAccount();
+
+    suiteSetup(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(120 * 1000);
+        oldAdvancedCreationSetting = <boolean>vscode.workspace.getConfiguration(extensionPrefix).get('advancedCreation');
+        await testAccount.signIn();
+        ext.tree = new AzureTreeDataProvider(FunctionAppProvider, 'azureFunctions.startTesting', undefined, testAccount);
+    });
+
+    suiteTeardown(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(1200 * 1000);
+        await vscode.workspace.getConfiguration(extensionPrefix).update('advancedCreation', oldAdvancedCreationSetting, vscode.ConfigurationTarget.Global);
+        const client: ResourceManagementClient = getResourceManagementClient(testAccount);
+        for (const resourceGroup of resourceGroupsToDelete) {
+            if (await client.resourceGroups.checkExistence(resourceGroup)) {
+                console.log(`Deleting resource group "${resourceGroup}"...`);
+                await client.resourceGroups.deleteMethod(resourceGroup);
+                console.log(`Resource group "${resourceGroup}" deleted.`);
+            } else {
+                // If the test failed, the resource group might not actually exist
+                console.log(`Ignoring resource group "${resourceGroup}" because it does not exist.`);
+            }
+        }
+        ext.tree.dispose();
+    });
+
+    test('createFunctionApp (basic)', async () => {
+        const resourceName1: string = getRandomHexString().toLowerCase();
```

Comment:
Naming the variable resourceName1 is kind of confusing since the other variable is just called resourceName.  Renaming them as resourceName1 and resourceName2 would be better.